components/renku_data_services/notebooks/core.py

```diff
@@ -288,6 +288,22 @@ async def docker_image_exists(config: NotebooksConfig, image_url: str, internal_
     return await image_repo.image_exists(parsed_image)
 
 
+async def docker_image_workdir(
+    config: NotebooksConfig, image_url: str, internal_gitlab_user: APIUser
+) -> PurePosixPath | None:
+    """Returns the working directory for the image.
+
+    If the user is logged in the internal GitLab (Renku V1), set the
+    credentials for the check.
+    """
+
+    parsed_image = Image.from_path(image_url)
+    image_repo = parsed_image.repo_api()
+    if parsed_image.hostname == config.git.registry and internal_gitlab_user.access_token:
+        image_repo = image_repo.with_oauth2_token(internal_gitlab_user.access_token)
```

Comment:
suggestion: there's one other place in code that does this check, but it's not really related to the workdir or image existence checks. it's more of an implementation detail. I'd move this logic somewhere more suited for it, e.g. directly in `repo_api()` or change the builder pattern to `maybe_with_oauth2` and pass it in there.

Reply:
Ok I added this in all places. Except in one where the for the v1 sessions it was important to distinguish whether the image is publicaly or pirvately available.